doc/src/devdocs/types.md

```diff
@@ -315,7 +315,129 @@ Ptr{Void} @0x00007f5998a04370
 
 so `Tuple == Tuple{Vararg{Any}}` is indeed the primary type.
 
-## Introduction to the internal machinery: `jltypes.c`
+## Diagonal types
+
+Consider the type `Tuple{T,T} where T`.
+A method with this signature would look like:
+
+```julia
+f(x::T, y::T) where T = ...
```

Comment:
I personally don't mind this how it is, but I imagine some will prefer `where {T}` to `where T`.